sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -108,6 +124,10 @@ export class EventHubProducer {
   ): Promise<void> {
     this._throwIfSenderOrConnectionClosed();
     throwTypeErrorIfParameterMissing(this._context.connectionId, "eventData", eventData);
+    if (Array.isArray(eventData) && eventData.length === 0) {
+      log.error(`[${this._context.connectionId}] Empty array was passed. No events to send.`);
+      return;
+    }
     if (eventData instanceof EventDataBatch && !eventData.batchMessage) {
```

Comment:
Now that we have count exposed on the batch, maybe we should check the count instead of `!eventData.batchMessage`. We can also initialize the `eventData.batchMessage` to an empty Buffer in the constructor for EventDataBatch and remove the `undefined` from its type. That way no one ever has to check for `!eventData.batchMessage`. 

Thoguhts?

Reply:
>Now that we have count exposed on the batch, maybe we should check the count instead of !eventData.batchMessage.

Agreed!

>We can also initialize the eventData.batchMessage to an empty Buffer in the constructor for EventDataBatch and remove the undefined from its type

It will not add any extra value, right?

Reply:
The extra value is that we dont have to check for `if (eventData.batchMessage)` or use `eventData.batchMessage!` before using that property

Reply:
If we initialize `batchMessage = Buffer.alloc();` in the EventDataBatch constructor, It will create a new buffer object of the specified size(in our case empty Buffer object).

Buffer object will use memory, so my point was why waste memory? If It is not adding any extra value other than this check `if (eventData.batchMessage)` or `eventData.batchMessage!`

Reply:
sounds good